go/proof.go

```diff
@@ -204,6 +204,14 @@ func (p *ExistenceProof) CheckAgainstSpec(spec *ProofSpec) error {
 	return nil
 }
 
+func keyForComparison(spec *ProofSpec, key []byte) []byte {
+	if !spec.PrehashCompareKey {
+		return key
+	}
+	hash, _ := doHashOrNoop(spec.LeafSpec.PrehashKey, key)
```

Comment:
Since this is using the PrehashKey defined in the leafspec and the key in the existenceProof is supposed to now be unhashed for SMT proofs, I think the SMT spec needs to be changed to have `PrehashKey=SHA_256` instead of `NO_HASH`

Reply:
I think the SMT should mirror the JMT's proof spec:

https://github.com/penumbra-zone/jmt/blob/main/src/tree/ics23_impl.rs#L219

Reply:
Right so the SMT proof spec in this repo need to be changed

Reply:
The code that generates the proofs will also need to be updated following the spec change